code/datums/status_effects/buffs.dm

```diff
@@ -925,3 +925,157 @@
 			var/obj/item/projectile/P = AM
 			if(P.flag == ENERGY || P.flag == LASER)
 				P.damage *= 0.85
+
+/datum/status_effect/flayer_rejuv
+	id = "rejuvination"
+	duration = 5 SECONDS
+	tick_interval = 1 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/flayer_rejuv
+	var/heal_amount = 5 // 25 total healing of both brute and burn at base
+
+/atom/movable/screen/alert/status_effect/flayer_rejuv
+	name = "Regenerating"
+	desc = "A ray of hope beyond disrepair."
+	icon_state = "drunk2"
+
+/datum/status_effect/flayer_rejuv/on_creation(mob/living/new_owner, extra_duration, extra_heal_amount)
+	if(isnum(extra_duration))
+		duration += extra_duration
+	if(isnum(extra_heal_amount))
+		heal_amount += extra_heal_amount
+	return ..()
+
+/datum/status_effect/flayer_rejuv/on_apply()
+	owner.SetWeakened(0)
+	owner.SetStunned(0)
+	owner.SetKnockDown(0)
+	owner.SetParalysis(0)
+	owner.SetSleeping(0)
+	owner.SetConfused(0)
+	owner.setStaminaLoss(0)
+	owner.stand_up(TRUE)
+	SEND_SIGNAL(owner, COMSIG_LIVING_CLEAR_STUNS)
+	return ..()
+
+/datum/status_effect/flayer_rejuv/tick()
+	if(!ishuman(owner))
+		return
+
+	var/mob/living/carbon/human/flayer = owner
+	flayer.adjustBruteLoss(-heal_amount, robotic = TRUE)
+	flayer.adjustFireLoss(-heal_amount, robotic = TRUE)
+	flayer.updatehealth()
+	if(flayer.has_status_effect(STATUS_EFFECT_TERMINATOR_FORM))
+		// Massive healing when in terminator mode
+		flayer.adjustStaminaLoss(-60)
+
+/datum/status_effect/quicksilver_form
+	id = "quicksilver_form"
+	duration = 10 SECONDS
+	tick_interval = 0
+	status_type = STATUS_EFFECT_REFRESH
+	alert_type = /atom/movable/screen/alert/status_effect/quicksilver_form
+	/// Temporary storage of the owner's flags to restore them properly after the ability is over
+	var/temporary_flag_storage
+	/// Do we also reflect projectiles
+	var/should_deflect = FALSE
+
+/atom/movable/screen/alert/status_effect/quicksilver_form
+	name = "Quicksilver body"
+	desc = "Your body is much less solid."
+	icon_state = "high"
+
+/datum/status_effect/quicksilver_form/on_creation(mob/living/new_owner, extra_duration, reflect_projectiles)
+	if(isnum(extra_duration))
+		duration += extra_duration
+	should_deflect = reflect_projectiles
+	return ..()
+
+/datum/status_effect/quicksilver_form/on_apply()
+	if(should_deflect)
+		ADD_TRAIT(owner, TRAIT_DEFLECTS_PROJECTILES, UNIQUE_TRAIT_SOURCE(src))
+	temporary_flag_storage = owner.pass_flags
+	owner.pass_flags |= (PASSTABLE | PASSGRILLE | PASSMOB | PASSFENCE | PASSGIRDER | PASSGLASS | PASSTAKE | PASSBARRICADE)
+	owner.add_atom_colour(COLOR_ALUMINIUM, TEMPORARY_COLOUR_PRIORITY)
+	return TRUE
+
+/datum/status_effect/quicksilver_form/on_remove()
+	if(should_deflect)
+		REMOVE_TRAIT(owner, TRAIT_DEFLECTS_PROJECTILES, UNIQUE_TRAIT_SOURCE(src))
```

Comment:
Just in case something happens with adminbus, I don't think anything bad can happen if we remove it regardless
```suggestion
	REMOVE_TRAIT(owner, TRAIT_DEFLECTS_PROJECTILES, UNIQUE_TRAIT_SOURCE(src))
```